test/unit/custom_spec.js

```diff
@@ -107,3 +107,91 @@ describe("custom canvas rendering", function () {
       .catch(done.fail);
   });
 });
+
+describe("alternate document context", function () {
+  const FontFace = global.FontFace;
+
+  let CanvasFactory;
+
+  beforeAll(() => {
+    global.FontFace = class {};
+    CanvasFactory = isNodeJS ? new NodeCanvasFactory() : new DOMCanvasFactory();
+  });
+
+  afterAll(() => {
+    global.FontFace = FontFace;
+  });
+
+  it("should use given document for loading fonts", async function () {
+    const domDoc = { fonts: new Set() };
+    const getDocumentParams = buildGetDocumentParams(
+      "TrueType_without_cmap.pdf",
+      {
+        disableFontFace: false,
+        ownerDocument: domDoc,
+      }
+    );
+
+    const doc = await getDocument(getDocumentParams).promise;
+
+    const page = await doc.getPage(1);
+
+    const viewport = page.getViewport({ scale: 1 });
+    const canvasAndCtx = CanvasFactory.create(viewport.width, viewport.height);
+
+    await page.render({
+      canvasContext: canvasAndCtx.context,
+      viewport,
+    }).promise;
+
+    expect(domDoc.fonts.size).toBe(1);
+
```

Comment:
You should also be destroying the `CanvasFactory` as well, both here and in the next test.